### PR TITLE
add glimmer/psionic artifact stuff

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
1 trigger, "Psionic disturbance":
- someone using a psionic power nearby, same as metapsionic popup

4 effects, "Noospheric shift":
- raise glimmer by 30 up to 450
- raise glimmer by 70 up to 700
- lower glimmer by 50 down to 400
- make everyone in a 5 meter radius psionic

## Why / Balance
glimmer and psionics have no real impact on research (turn off prober roundstart and you are basically safe forever, psionics have no integration with research at all)

this makes psionic powers a trigger so having people be psionic is a necessary evil for epi, and an effect that can sometimes make them psionic
raising glimmer can be used by traitors, if you can get it in a loop and farm it you can permanently keep glimmer at a moderate or high level :trollface:
lowering glimmer to a safe level can be used in an emergency 

## Technical details
- optimised and refactored metapsionic logic so it only looks up metapsionic power instead of any entity in 10 meters
- fixed only 1 person with metapsionic pulse being notified of a power being used, now everyone with it gets notified

## Media
a lower glimmer node

psionic effect is real

## Requirements
- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- add: Artifacts can now interact with the noosphere in several ways.

### CODE REVIEW

File: Content.Server/DeltaV/Xenoarchaeology/XenoArtifacts/Effects/Components/PsionicProducingArtifactComponent.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    /// Range to look for potential psionics in.
+    /// </summary>
+    [DataField(required: true)]
+    public float Range;
+
+    /// <summary>
+    /// Number of times this node can trigger before it switches to doing nothing.
+    /// </summary>
+    [DataField]
+    public int Limit = 1;
```
Comment (Samsterious) [inline Content.Server/DeltaV/Xenoarchaeology/XenoArtifacts/Effects/Components/PsionicProducingArtifactComponent.cs:21]: One of my first thoughts was that this was a little too powerful without at least a "_low chance_ of making everyone in X radius a psychic", but this works too!

File: Resources/Locale/en-US/deltav/xenoarchaeology/artifact-hints.ftl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# effects
+
+artifact-effect-hint-noosphere = Noospheric shift
```
Comment (Samsterious) [inline Resources/Locale/en-US/deltav/xenoarchaeology/artifact-hints.ftl]: _Noöspheric_ shift 😜

File: Resources/Prototypes/DeltaV/XenoArch/Effects/glimmer.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+- type: artifactEffect
+  id: EffectRaiseGlimmer
+  targetDepth: 2
+  effectHint: artifact-effect-hint-noosphere
+  components:
+  - type: GlimmerArtifact
+    range:
+      min: 0
+      max: 450 # cant instantly get ended by a looping artifact
```
Comment (Samsterious) [inline Resources/Prototypes/DeltaV/XenoArch/Effects/glimmer.yml:9]: Thank you for this 🙏

**Review by ghost — COMMENTED** [review]
YAML / FTL looks fine - if I can get someone's eyes on the C# to also say it looks fine that would be nice. @MilonPL @MilonPL @MilonPL

**Comment by deltanedas** [comment]
> YAML / FTL looks fine

:trollface:

**Review by TadJohnson00 — APPROVED** [review]
why was I requested for review :imcry:
this all looks fine from yaml but need someone to check over C# before merge

**Comment by Samsterious** [comment]
> why was I requested for review :imcry:

Because you're leading the charge on psionics and I wanted your opinion, nerd :blunt:

**Comment by TadJohnson00** [comment]
> > why was I requested for review :imcry:
> 
> Because you're leading the charge on psionics and I wanted your opinion, nerd :blunt:

just throw a book or a blunt object at me over discord nerd!!!

**Review by ghost — APPROVED** [review]
I didn't catch that, reapproving though.

**Pull request merged**